flowkit_jwt_generator/flowkit_jwt_generator/jwt.py

```diff
@@ -1,24 +1,109 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import base64
 import datetime
 import os
 import uuid
+from binascii import Error
 from json import JSONEncoder
-from typing import Dict, List, Union, Callable
+from typing import Dict, List, Union, Callable, Tuple, Optional
 from collections import ChainMap
 from datetime import timedelta
 import click
 import jwt
 import pytest
 import requests
-
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.backends.openssl.rsa import _RSAPrivateKey, _RSAPublicKey
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
 
 permissions_types = {"run": True, "poll": True, "get_result": True}
 aggregation_types = ["admin0", "admin1", "admin2", "admin3", "admin4"]
 
 
+def load_private_key(key_string: str) -> _RSAPrivateKey:
+    """
+    Load a private key from a string, which may be base64 encoded.
+
+    Parameters
+    ----------
+    key_string : str
+        String containing the key, optionally base64 encoded
+
+    Returns
+    -------
+    _RSAPrivateKey
+        The private key
+    """
+    try:
+        return serialization.load_pem_private_key(
+            key_string.encode(), password=None, backend=default_backend()
+        )
+    except ValueError:
+        try:
+            return load_private_key(base64.b64decode(key_string).decode())
+        except (Error, ValueError):
```

Comment:
Again nitpick, but reading this at first I thought we're catching generic exceptions until I realised that `Error` was imported from `binascii`. How about making this explicit here so that it's clear _why_ the command might fail? (Requires tweaking the import, of course.)
```suggestion
        except (binascii.Error, ValueError):
```